frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx

```diff
@@ -30,50 +41,257 @@ export default function DiscoveredByCluster({
+        {
+          id: 'deploymentAvailable',
+          label: t('Deployment available'),
+          options: [
+            {
+              label: t('yes'),
+              value: 'yes',
+            },
+            {
+              label: t('no'),
+              value: 'no',
+            },
+          ],
+          tableFilterFn: (selectedValues: string[], item: DiscoveredPolicyItem): boolean => {
+            return selectedValues.includes(item.deploymentAvailable ? 'yes' : 'no')
+          },
+        },
+        {
+          id: 'upgradeAvailable',
+          label: t('Upgrade available'),
+          options: [
+            {
+              label: t('yes'),
+              value: 'yes',
+            },
+            {
+              label: t('no'),
+              value: 'no',
+            },
+          ],
+          tableFilterFn: (selectedValues: string[], item: DiscoveredPolicyItem): boolean => {
+            return selectedValues.includes(item.upgradeAvailable ? 'yes' : 'no')
+          },
+        },
+      ])
+    }
+
+    return filters
+  }, [t, policyKind, policies])
+
   return (
-    <PageSection>
+    <React.Fragment>
       {policies && policies.length > 0 && (
-        <Box mb={3} maxWidth={500}>
-          <DiscoveredViolationsCard
-            policyKind={policyKind}
-            policyViolationSummary={policyViolationSummary(policies ?? [])}
-          />
-        </Box>
+        <PageSection style={{ paddingBottom: '0' }}>
+          <AcmMasonry minSize={450} maxColumns={3}>
+            <DiscoveredViolationsCard
+              policyKind={policyKind}
+              policyViolationSummary={policyViolationSummary(policies ?? [])}
+            />
+            {policyKind === 'OperatorPolicy' && (
+              <AcmDonutChart
+                title={t('Deployments unavailable')}
+                description={t('Overview of unavailable deployments')}
+                donutLabel={{
+                  title: operatorPolicyStats['deploymentsUnavailable'],
+                  subTitle: t('Unavailable'),
+                }}
+                data={[
+                  {
+                    key: t('unavailable'),
+                    value: operatorPolicyStats['deploymentsUnavailable'],
+                    isPrimary: true,
+                    useForTitleCount: true,
+                    link: locationPath + '?deploymentAvailable=no',
+                  },
+                  {
+                    key: t('with no status'),
+                    value: operatorPolicyStats['deploymentsUnknown'],
+                  },
+                  {
+                    key: t('available'),
+                    value: operatorPolicyStats['deploymentsAvailable'],
+                    link: locationPath + '?deploymentAvailable=yes',
+                  },
+                ]}
+                colorScale={colorThemes.criticalLowSuccess}
+              />
+            )}
+            {policyKind === 'OperatorPolicy' && (
+              <AcmDonutChart
```

Comment:
We can put this to line 217 together. Wrap with `<></>`

Reply:
@yiraeChristineKim I tried this but it adds an additional div which causes both donut charts to be in the same grid column.

Reply:
Matt try this 
```
diff --git a/frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx b/frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx
index 28ef159d..2b9e0b6e 100644
--- a/frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx
+++ b/frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx
@@ -1,11 +1,9 @@
 /* Copyright Contributors to the Open Cluster Management project */
 import { PageSection } from '@patternfly/react-core'
-import { Card } from '@patternfly/react-core'
 import { DiscoveredPolicyItem } from '../useFetchPolicies'
 import { AcmDonutChart, AcmEmptyState, AcmTable, ITableFilter, colorThemes } from '../../../../ui-components'
 import { useTranslation } from '../../../../lib/acm-i18next'
 import { useRecoilValue, useSharedAtoms } from '../../../../shared-recoil'
-import { AcmMasonry } from '../../../../components/AcmMasonry'
 import {
   byClusterCols,
   convertYesNoCell,
@@ -18,6 +16,7 @@ import {
 } from './common'
 import { useMemo } from 'react'
 import { useLocation } from 'react-router-dom-v5-compat'
+import { Grid } from '@mui/material'
 
 export default function DiscoveredByCluster({
   policies = [],
@@ -206,70 +205,76 @@ export default function DiscoveredByCluster({
     <>
       {policies && policies.length > 0 && (
         <PageSection style={{ paddingBottom: '0' }}>
-          <AcmMasonry minSize={450} maxColumns={3}>
-            <DiscoveredViolationsCard
-              policyKind={policyKind}
-              policyViolationSummary={policyViolationSummary(policies ?? [])}
-            />
-            {policyKind === 'OperatorPolicy' && (
-              <AcmDonutChart
-                title={t('Deployments unavailable')}
-                description={t('Overview of unavailable deployments')}
-                donutLabel={{
-                  title: operatorPolicyStats['deploymentsUnavailable'],
-                  subTitle: t('Unavailable'),
-                }}
-                data={[
-                  {
-                    key: t('unavailable'),
-                    value: operatorPolicyStats['deploymentsUnavailable'],
-                    isPrimary: true,
-                    useForTitleCount: true,
-                    link: locationPath + '?deploymentAvailable=no',
-                  },
-                  {
-                    key: t('with no status'),
-                    value: operatorPolicyStats['deploymentsUnknown'],
-                  },
-                  {
-                    key: t('available'),
-                    value: operatorPolicyStats['deploymentsAvailable'],
-                    link: locationPath + '?deploymentAvailable=yes',
-                  },
-                ]}
-                colorScale={colorThemes.criticalLowSuccess}
+          <Grid container spacing={2}>
+            <Grid item>
+              <DiscoveredViolationsCard
+                policyKind={policyKind}
+                policyViolationSummary={policyViolationSummary(policies ?? [])}
               />
-            )}
+            </Grid>
             {policyKind === 'OperatorPolicy' && (
-              <AcmDonutChart
-                title={t('Upgrade availability')}
-                description={t('Overview of available upgrades')}
-                donutLabel={{
-                  title: operatorPolicyStats['upgradesAvailable'],
-                  subTitle: t('Available'),
-                }}
-                data={[
-                  {
-                    key: t('available'),
-                    value: operatorPolicyStats['upgradesAvailable'],
-                    isPrimary: true,
-                    useForTitleCount: true,
-                    link: locationPath + '?upgradeAvailable=yes',
-                  },
-                  {
-                    key: t('with no status'),
-                    value: operatorPolicyStats['upgradesUnknown'],
-                  },
-                  {
-                    key: t('unavailable'),
-                    value: operatorPolicyStats['upgradesUnavailable'],
-                    link: locationPath + '?upgradeAvailable=no',
-                  },
-                ]}
-                colorScale={colorThemes.criticalLowSuccess}
-              />
+              <>
+                <Grid item>
+                  <AcmDonutChart
+                    title={t('Deployments unavailable')}
+                    description={t('Overview of unavailable deployments')}
+                    donutLabel={{
+                      title: operatorPolicyStats['deploymentsUnavailable'],
+                      subTitle: t('Unavailable'),
+                    }}
+                    data={[
+                      {
+                        key: t('unavailable'),
+                        value: operatorPolicyStats['deploymentsUnavailable'],
+                        isPrimary: true,
+                        useForTitleCount: true,
+                        link: locationPath + '?deploymentAvailable=no',
+                      },
+                      {
+                        key: t('with no status'),
+                        value: operatorPolicyStats['deploymentsUnknown'],
+                      },
+                      {
+                        key: t('available'),
+                        value: operatorPolicyStats['deploymentsAvailable'],
+                        link: locationPath + '?deploymentAvailable=yes',
+                      },
+                    ]}
+                    colorScale={colorThemes.criticalLowSuccess}
+                  />
+                </Grid>
+                <Grid item>
+                  <AcmDonutChart
+                    title={t('Upgrade availability')}
+                    description={t('Overview of available upgrades')}
+                    donutLabel={{
+                      title: operatorPolicyStats['upgradesAvailable'],
+                      subTitle: t('Available'),
+                    }}
+                    data={[
+                      {
+                        key: t('available'),
+                        value: operatorPolicyStats['upgradesAvailable'],
+                        isPrimary: true,
+                        useForTitleCount: true,
+                        link: locationPath + '?upgradeAvailable=yes',
+                      },
+                      {
+                        key: t('with no status'),
+                        value: operatorPolicyStats['upgradesUnknown'],
+                      },
+                      {
+                        key: t('unavailable'),
+                        value: operatorPolicyStats['upgradesUnavailable'],
+                        link: locationPath + '?upgradeAvailable=no',
+                      },
+                    ]}
+                    colorScale={colorThemes.criticalLowSuccess}
+                  />
+                </Grid>
+              </>
             )}
-          </AcmMasonry>
+          </Grid>
         </PageSection>
       )}
       <PageSection>

```